packages/actions/src/internal/runCallWithPrestateTrace.js

```diff
@@ -110,23 +110,63 @@ const captureAccountState = async (vm, logger, address, slots = new Set()) => {
 		for (const slotHex of slots) {
 			try {
 				// Convert hex slot to Uint8Array for getContractStorage
-				const slotKey = Buffer.from(slotHex.slice(2), 'hex')
+				const slotKey = hexToBytes(slotHex)
 				const value = await vm.stateManager.getContractStorage(address, slotKey)
 				storage[slotHex] = bytesToHex(value)
 			} catch (err) {
 				logger.error(err, `Error getting storage at slot ${slotHex} for account ${address.toString()}`)
 			}
 		}
 
-		return {
+		/** @type {{-readonly [K in keyof import('../debug/DebugResult.js').AccountState]: import('../debug/DebugResult.js').AccountState[K]}} */
+		const state = {
+			storage,
 			balance: toHex(account.balance),
 			nonce: account.nonce.toString(),
 			code: code && code.length > 0 ? bytesToHex(code) : '0x',
-			storage,
+			codeHash: undefined,
+			codeSize: undefined,
+			storageRoot: undefined,
+			isContract: undefined,
+			isEmpty: undefined,
+			version: undefined,
+		}
+
+		try {
+			state.codeHash = bytesToHex(account.codeHash)
+		} catch (err) {
+			logger.error(err, `code hash not loaded for account ${address.toString()}`)
+		}
```

Comment:
generally not a good practice to use thrown errors in the happy path control flow. We want to throw an error if for example the codeHash is not valid bytes and this will swallow it.
```suggestion
if (account.codeHash !== undefined) {
		state.codeHash = bytesToHex(account.codeHash)
}
```